tests/test_other.py

```diff
@@ -9698,6 +9698,24 @@ def test_add_emscripten_metadata(self):
     ret = run_process(NODE_JS + ['hello_world.js'], stdout=PIPE).stdout
     self.assertTextDataIdentical('hello, world!\n', ret)
 
+  @parameterized({
+    'O0': (False, []), # noqa
+    'O2': (False, ['-O2']), # noqa
+    'O2_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE']), # noqa
+    'O2_js_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '-s', 'WASM=0']), # noqa
+    'O2_closure_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1']), # noqa
+    'O2_closure_js_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1', '-s', 'WASM=0']), # noqa
```

Comment:
Give that you are testing at different O levels would this make sense in core?

Reply:
Yeah, maybe this could go in core. I kind of like explicitly saying what to test in this case though, since it intentionally includes closure. We don't have a core test mode for closure, so we'd need to add it in manually into core, which removes most of the benefit there I think.